lib/spack/spack/directory_layout.py

```diff
@@ -260,9 +260,10 @@ def check_installed(self, spec):
             return None
 
         if not os.path.isfile(spec_file_path):
-            raise InconsistentInstallDirectoryError(
-                'Install prefix exists but contains no spec.yaml:',
-                "  " + path)
+            return None
+#            raise InconsistentInstallDirectoryError(
+#                'Install prefix exists but contains no spec.yaml:',
+#                "  " + path)
```

Comment:
> Why exactly is this change necessary?

It fixes #2319 

> Is it only necessary when running spack setup?

Yes

> Could we perhaps write a spec.yaml for packages installed by spack setup?

I'm sure there's a better way to fix #2319.  Please see full notes above this comment on the GitHub discussion page (not just in email).